generator/src/test/java/com/scottlogic/deg/generator/generation/FieldSpecGroupValueGeneratorTest.java

```diff
@@ -0,0 +1,91 @@
+package com.scottlogic.deg.generator.generation;
+
+import com.scottlogic.deg.common.profile.Field;
+import com.scottlogic.deg.generator.fieldspecs.FieldSpec;
+import com.scottlogic.deg.generator.fieldspecs.FieldSpecGroup;
+import com.scottlogic.deg.generator.fieldspecs.relations.FieldSpecRelations;
+import com.scottlogic.deg.generator.fieldspecs.whitelist.DistributedSet;
+import com.scottlogic.deg.generator.fieldspecs.whitelist.FrequencyDistributedSet;
+import com.scottlogic.deg.generator.generation.databags.DataBag;
+import com.scottlogic.deg.generator.generation.databags.DataBagValue;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class FieldSpecGroupValueGeneratorTest {
+
+    @Test
+    public void testGenerationOneField() {
```

Comment:
I think it would be clearer to call this test `generate_withGroupOfSingleField_returnsCorrectStream` or similar. See also the other test below.